### PR TITLE
Added Example set_vm_metadata Integration Test

### PR DESCRIPTION
**Pull request by posijon**
`integration/set_vm_metadata_test` uses `set_vm_metadata` method to setTags on a SoftLayer VM. This involved adding templates in `test_fixtures/cpi_methods` that the tests modify. Also see `test_helpers` for helper functions and more unit tests.

Paired with @maximilien

### CODE REVIEW

File: integration/virtual_guest_tags/virtual_guest_tags_test.go [hunk 1]
```diff
@@ -0,0 +1,136 @@
+/*
+Integration test for tags
```
Comment (maximilien) [inline integration/virtual_guest_tags/virtual_guest_tags_test.go]: Do we need this file? Seems like it should be deleted?

Comment (posijon) [inline integration/virtual_guest_tags/virtual_guest_tags_test.go]: I deleted that folder in a later commit.

File: test_helpers/integration_helpers_test.go [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+			Expect(err).ToNot(HaveOccurred())
+			Expect(fileInfo.Mode().IsRegular()).To(BeTrue())
+
+			err = os.RemoveAll(tmpConfigPath)
+			Expect(err).ToNot(HaveOccurred())
+		})
+	})
+
+	Context("#GenerateCpiJsonPayload", func() {
+		//TODO: Context("error cases", ...)
```
Comment (maximilien) [inline test_helpers/integration_helpers_test.go]: Let's either implement these tests now or remove the TODO and move them as one issue in the repo and chore on Tracker.

**Comment by maximilien** [comment]
Please address one comment and send again and I'll merge when Travis is green. Thanks.

**Comment by posijon** [comment]
Need to implement has_vm tests as per issue #35.

**Comment by posijon** [comment]
Also see tracker ID [#97163142](https://www.pivotaltracker.com/story/show/97163142)

**Pull request merged**